### PR TITLE
CAMEL-14994: Improve Projects Page

### PR DESCRIPTION
**Pull request by aashnajena**
First draft for improving the projects page. I'm opening the PR so that the preview is available. Need feedback on a couple of things :
- I've taken icons from Eva, from where icons in the "Packed with Functionality"  section (on front page) are taken. Do the outlined icons look better or the filled ones?
- Do we need to add text explaining the icons?
- As pointed out on the JIRA issue, the mailing list is common for all projects, so should we include that in each project box?
- As pointed out on the JIRA issue, we should be adding links to examples. What's the best way to add those?
- Quarkus has a separate Gitter chat. Do other projects have this too? We can add links to those.

Also, in <1024 screens, the h1 headings were becoming smaller than other headings, so I removed that.

### CODE REVIEW

**Comment by zregvart** [comment]
There's a comment on the [issue](https://issues.apache.org/jira/browse/CAMEL-14994) to include the examples. In addition I think we should have text labels along with icons to make the touch targets a bit bigger on smaller screens and to make the distinction between the source and example repositories clearer.

The projects.css rules need to be scoped to `.projects` and it would be better to use semantic markup where applicable.

**Comment by aashnajena** [comment]
> There's a comment on the [issue](https://issues.apache.org/jira/browse/CAMEL-14994) to include the examples. In addition I think we should have text labels along with icons to make the touch targets a bit bigger on smaller screens and to make the distinction between the source and example repositories clearer.
> 
> The projects.css rules need to be scoped to `.projects` and it would be better to use semantic markup where applicable.

**Comment by aashnajena** [comment]
Will something like this make more sense? (The icons need to be changed, their positions need to be adjusted, this is just draft). Buttons make for a much better touch target.

image

**Comment by davsclaus** [comment]
Yeah nice with icon and a little text

- Docs
- Examples
- Source

**Comment by aashnajena** [comment]
Please review the changes. Also, are there any changes needed in the projects section on the front-page? Since all important links are on this page now, I think it would be good if we link the projects displayed on the front page to their corresponding entries on the "Projects" page. 

On the front page, instead of providing description+links, we can just have the logo, one-line description, and make it clickable so that it links to the projects page. I think having a link to just "documentation" on the front page makes it look like that's the only important link.

**Comment by zregvart** [comment]
What I’ve noticed Is that in the mobile version the links overlap.

**Comment by aashnajena** [comment]
What's the smallest width for which we need to adjust our design? Right now, the buttons get broken at <400px width, so I will fix that.

**Comment by zregvart** [comment]
@aashnajena I test on my phone, which has 375 points width (750 pixels), when testnih in the browser I test with Samsung Galaxy 6/9 which has 360 points (IIRC).

Here are some statistics to consider:
https://gs.statcounter.com/screen-resolution-stats/mobile/worldwide

File: static/img/icons/book-icon.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><g data-name="Layer 2"><g data-name="book"><rect width="24" height="24" opacity="0"/><path d="M19 3H7a3 3 0 0 0-3 3v12a3 3 0 0 0 3 3h12a1 1 0 0 0 1-1V4a1 1 0 0 0-1-1zM7 5h11v10H7a3 3 0 0 0-1 .18V6a1 1 0 0 1 1-1zm0 14a1 1 0 0 1 0-2h11v2z"/></g></g></svg>
```
Comment (zregvart) [inline static/img/icons/book-icon.svg:1]: Images need to be placed in the `antora-ui-camel/src/img` directory, we have a pipeline for optimizing SVGs at build time.

File: antora-ui-camel/src/css/projects.css [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  padding: 0.5rem;
+  margin: 0 0.4rem;
+  border: 1px solid var(--color-gray-10);
+  border-radius: 10rem;
+  color: var(--body-font-color);
+}
+
+.projects .box .links span:hover {
+  color: var(--body-font-color);
+  border: 2px solid var(--color-gray-30);
```
Comment (zregvart) [inline antora-ui-camel/src/css/projects.css:54]: This changes the metrics of the link and causes a reflow.

File: antora-ui-camel/src/css/projects.css [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+  .projects .box {
+    padding: 1rem 0.4rem;
+  }
+
+  .projects .box .links img {
+    height: 0.8rem;
+    width: 0.8rem;
+    vertical-align: -0.1rem;
+    margin: 0 0.2rem;
```
Comment (zregvart) [inline antora-ui-camel/src/css/projects.css:86]: This looks very odd, we should a simpler and more straightforward way to center images.

**Comment by zregvart** [comment]
@aashnajena sounds great, I'd like to unify the design across the front page and the projects page, I suggest that we start with the simplest design and work from that. I'm really adverse to shortening the words, e.g. documentation to docs, this is my personal preference, doesn't have to be universally shared.

I strongly believe that making things simpler takes more effort and that it pays of in the future by making maintenance/changes easier to do. We need to have a consistent design, I suggest that we start with the minimal changes and unify/change design between the front page and the projects page.

File: antora-ui-camel/src/css/doc.css [hunk 4]
```diff
@@ -652,11 +652,6 @@ kbd,
 }
 
 @media screen and (max-width: 1023px) and (min-width: 480px) {
-  .doc > h1.page:first-child,
-  .static > h1:first-child {
-    font-size: inherit;
-  }
-
```
[the remark this answers is not in the document]
Comment (aashnajena) [reply antora-ui-camel/src/css/doc.css:659]: This piece of code is causing h1's font-size to become smaller than other headings (only for 480px-1023px). 

image

**Comment by aashnajena** [comment]
@zregvart Okay, here's what I think : On the rest of the website, links are never presented as buttons, and I don't see icons associated with links either. I think it's best to keep the project card simple like shown below : 

projects-design-simple

I feel that we're repeating too much information on the front page, like description and links. When I visit the front page and see the projects section, I think that documentation is the only available link for each project. I propose that we make the front page "projects" section much simpler, with logos, heading and one-line description. This is completely reusing the design from the projects page.

frontpage-projects-simple

If the user wants to read more about this project, they can click on the card, which can take them to the projects page. Right now, clicking on the logo takes you to the manual for the project. I think we can maintain the hierarchy as frontpage-project-card (small description) --> projects-section project card (longer description + useful links) --> project manuals (detailed documentation). 

Would we prefer this or do we want links on the front page as well?

**Comment by aashnajena** [comment]
Pushed changes which resolve/undo most of the issues mentioned above, and simplify the front-page.  @zregvart @davsclaus Please go through how the front-page and projects-page look for different screen sizes now. If we want to restore the bigger descriptions/ links on front-page, it wouldn't be hard.

**Comment by zregvart** [comment]
This also needs to be rebased on latest changes in master branch.

**Comment by aashnajena** [comment]
I'm not using the icons in this iteration, since links are not accompanied by icons anywhere else on the website. I'll delete the icons.

**Comment by oscerd** [comment]
Build is failing after merging this

11:03:47  $ antora --generator @antora/xref-validator antora-playbook.yml
11:06:11  $ link-checker --disable-external --mkdocs --allow-hash-href public --url-ignore='#top'
11:07:07  anchor not found from index.html to projects#camel
11:07:07  anchor not found from index.html to projects#camel-k
11:07:07  anchor not found from index.html to projects#camel-kafka-connector
11:07:07  anchor not found from index.html to projects#camel-spring-boot
11:07:07  anchor not found from index.html to projects#camel-karaf

https://builds.apache.org/view/C/view/Apache%20Camel/job/Camel.website/job/master/236/console

**Comment by aashnajena** [comment]
What's the workaround for this? The redirects work fine on the preview

**Comment by oscerd** [comment]
There shouldn't be a workaround, we need to always build the website. This should fail OOTB while building.

**Comment by aashnajena** [comment]
Running `yarn checks` locally isn't giving me any errors under `$ link-checker`. How can I correct these links?

**Comment by oscerd** [comment]
There is a PR from @zregvart

**Comment by zregvart** [comment]
This should be fixed with 855259a14f2a1cf10f52c19bd2635f5e98ac24be, I think the link checker should report errors. Perhaps it's the `CAMEL_ENV=production` which triggers minification that introduces the broken links. Anyhow the correct way to link to `index.html` pages is either by specifying the `/path/index.html` or to put the trailing slash on the path, e.g. `/path/`. That only works because Apache tolerates both, I think there's an extra roundtrip with a redirect, so this comes at a cost.

**Pull request merged**